pep/app/pods/most-cited/controller.ts

```diff
@@ -7,30 +7,48 @@ import { inject as service } from '@ember/service';
 import LoadingBarService from 'pep/services/loading-bar';
 import FastbootMediaService from 'pep/services/fastboot-media';
 import SidebarService from 'pep/services/sidebar';
-
-type PossiblePeriodValues = '5' | '10' | '20' | 'all';
-type period = { label: string; value: PossiblePeriodValues };
+import IntlService from 'ember-intl/services/intl';
+import Journal from 'pep/pods/journal/model';
+import { PERIODS, PossiblePeriodValues } from 'pep/constants/sidebar';
 
 export default class MostCited extends Controller {
     @service loadingBar!: LoadingBarService;
     @service fastbootMedia!: FastbootMediaService;
     @service sidebar!: SidebarService;
+    @service intl!: IntlService;
 
     queryParams = ['author', 'title', 'sourcename', 'period'];
 
     @tracked paginator!: Pagination<Document>;
     @tracked author = '';
     @tracked title = '';
-    @tracked sourcename = '';
+    @tracked journal?: Journal;
     @tracked period: PossiblePeriodValues = 'all';
+
+    /**
+     * GET/SET for sourcename. Ember requires the set if we are using it as a query param.
+     * Eventually we should use journal IDS to search on instead of the title so that way we can
+     * reload them if coming from a pasted link
+     *
+     * @memberof MostCited
+     */
+    get sourcename() {
+        return this.journal?.title ?? '';
+    }
+    set sourcename(value: string) {
+        console.log(value);
```

Comment:
i would just comment out the console.log if this isnt fully done yet

Reply:
Leaving this in to allow for the green builds you mentioned